src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,32 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier extends ColorSupplier {
+    public static final int MAX_NUMBER_FIGURES = 6;
+    public static final int MAX_NUMBER_SIDE = 20;
+    private static final Random random = new Random();
+    private static final ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure(String randomColor) {
+
+        int shape = random.nextInt(MAX_NUMBER_FIGURES);
+        int sideNew = random.nextInt(MAX_NUMBER_SIDE);
+        String color = String.valueOf(colorSupplier.getRandomColor());
+        if (shape == 0) {
+            return new Square(color, sideNew);
```

Comment:
```suggestion
        if (shape == 0) {
            return getRandomSquare();
```
lets add private methods to get each figure - this makes your code easier to read